### PR TITLE
List views

### LINKED ISSUE #402: List view [$45]
Please add an agenda view to the nextcloud calendar. In case you don't know, what I'm referring to, an agenda view is basically a list of the upcoming events in your calendar.

The view button could be placed right beside the day view, week view, month view buttons, e.g.

Of course, this view type should also be available for public calendars.

<bountysource-plugin>

---
There is a **[$45 open bounty](https://www.bountysource.com/issues/43719956-list-view?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github)** on this issue. Add to the bounty at [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F45525646&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by jean-gui**
## Description

This PR adds list views per week and month in addition to the day/week/month views. These views are also available to publicly shared calendars. Fortunately, FC5 already supports these views so there wasn't much to do.

Fixes #402 

Note that these views only show events for the currently selected week/month. For example, if there's an event on 2020-08-19 (Wednesday) and another one on 2020-08-24 (Monday), only one will be displayed in the week list at once even though there's only 5 days separating them. This seems to be how FC5 works.

Also note that I literally learned about NC, FC and JS/Vue development just for this so I basically replicated what already existed for the other views without really trying to understand how they worked, and I may not be able to add more features.

### Type of change

- [X] New feature (non-breaking change which adds functionality)

### How to test / use your changes?

Please provide instructions how to test your changes

1. Install this version of the app. A tar.gz created with `make build-js-production appstore` is temporarily available at https://cloud.troulite.fr/index.php/s/8MniantZGyxod52
2. Got to your calendars
3. Open the view selector, there should be two new views (Week (list) and Month (list))
4. Clicking on one of them opens the corresponding view

### UI Changes

In case you changed, added or removed UI elements, please provide a set of before / after screenshots:

| Before | After |
|:---------:|:------:|
|before1|after1|
|before2|after2|

### Checklist:

- [X] My code follows the style guidelines of this project
- [X] I signed off my changes (`git commit -sm "Your commit message"`)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas (since this is mostly a duplication of existing code I don't think there's a need for more comments)
- [ ] I have made corresponding changes to the documentation (I don't think there's a need to change the documentation)
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (I don't know if this needs tests nor how to add them)
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by georgehrke** [comment]
@jean-gui Thanks a lot for your pull-request!
We just merged a PR updating FullCalendar to 5.3 a couple of days ago.

Can you please rebase your pull-request?

Also it does not show the location / description. 
If space allows, at least the first two lines of location / description should be shown.

File: lib/RepairSteps/CurrentViewNameRepairStep.php [hunk 1]
```diff
@@ -85,6 +85,14 @@ public function run(IOutput $output):void {
 					$this->config->setUserValue($userId, 'calendar', 'currentView', 'timeGridWeek');
 					break;
 
+				case 'listWeek':
```
Comment (georgehrke) [inline lib/RepairSteps/CurrentViewNameRepairStep.php:88]: This file is only a migration step for users coming from Calendar 1.x.

Version 1 had no list view, so there is no need to change the saved view.
Please remove lines 88 to 95.

File: src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue [hunk 2]
```diff
@@ -55,6 +55,14 @@ export default {
 				id: 'dayGridMonth',
 				icon: 'icon-view-module',
 				label: this.$t('calendar', 'Month'),
+			}, {
+				id: 'listWeek',
```
Comment (georgehrke) [inline src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue:59]: There shouldn't be two list views. Please remove `listWeek` and keep `listMonth`.

**Review by georgehrke — CHANGES_REQUESTED** [review]

File: src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue (hunk 2 above)
Comment (jean-gui) [reply src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue:59]: Why? Do you think the list per week is unnecessary? If so, should I remove mentions to it from everywhere in the code?

File: lib/RepairSteps/CurrentViewNameRepairStep.php (hunk 1 above)
Comment (jean-gui) [reply lib/RepairSteps/CurrentViewNameRepairStep.php]: Done.

File: src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue (hunk 2 above)
Comment (georgehrke) [reply src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue:63]: > Why? Do you think the list per week is unnecessary?

Yes, it does not have any real advantage over list month. The information displayed and the presentation is identical.

Comment (jean-gui) [reply src/components/AppNavigation/AppNavigationHeader/AppNavigationHeaderViewMenu.vue]: Seems fair, done.

**Comment by jean-gui** [comment]
> @jean-gui Thanks a lot for your pull-request!
> We just merged a PR updating FullCalendar to 5.3 a couple of days ago.
> 
> Can you please rebase your pull-request?
> 
> Also it does not show the location / description.
> If space allows, at least the first two lines of location / description should be shown.

Done. I think a change in an html class had broken location. I have now also added the description. If it's longer than two lines, it adds '...' after those first two lines.

**Comment by georgehrke** [comment]
@jean-gui I hope you don't mind, but i went ahead and pushed on your branch. There have been a couple of minor things and i really want to include this in 2.1.

| Light | Dark |
|:---------:|:------:|
|empty_light|empty_dark|
|list_light|list_dark|

**Comment by georgehrke** [comment]
I also removed the translations you added. Our translations are managed via http://transifex.com/ and automatically synced here via a cron-job. Every manual change to `l10n/` will be lost after the cron-job.

**Comment by jean-gui** [comment]
> @jean-gui I hope you don't mind, but i went ahead and pushed on your branch. There have been a couple of minor things and i really want to include this in 2.1.

No problem at all

> I also removed the translations you added. Our translations are managed via http://transifex.com/ and automatically synced here via a cron-job. Every manual change to `l10n/` will be lost after the cron-job.

Will you add the new string to transifex? If you tell me how to do so, I can add the French translation.

**Comment by georgehrke** [comment]
> Will you add the new string to transifex? If you tell me how to do so, I can add the French translation.

I will do that in the next couple of days, but our translation community is so fast, so I'm pretty sure they will be faster than me.

**Pull request merged**